pointpats/centrography.py

```diff
@@ -590,95 +590,122 @@ def _(points: GeoPandasBase) -> np.float64:
     return std_distance(coords)
 
 
+
 @singledispatch
-def ellipse(points):
+def ellipse(points, weights=None, method="crimestat", 
+              crimestatCorr=True, degfreedCorr=True):
+    """
+    Computes a weighted standard deviational ellipse for a set of point geometries.
+
+    Parameters:
+        points : arraylike
+                 array representing a point pattern
+        weights : arraylike
+                 array of weights for each point (Optional)
+        method (str): Correction method to apply. Must be 'crimestat' or 'yuill'.
+        crimestatCorr (bool): Apply CrimeStat correction if method == 'yuill'.
+        degfreedCorr (bool): Apply degrees-of-freedom correction if method == 'yuill'.
+
+    Returns:
+        tuple: (major_axis_length, minor_axis_length, rotation_angle)
     """
-    Calculate parameters of standard deviational ellipse for a point pattern.
-
-    Parameters
-    ----------
-    points : arraylike
-             array representing a point pattern
-
-    Returns
-    -------
-    ellipse
-        representation of the standard ellipse
-
-    Notes
-    -----
-    Implements approach from:
-
-    https://www.icpsr.umich.edu/CrimeStat/files/CrimeStatChapter.4.pdf
-
-    Examples
```

Comment:
Any chance of keeping the examples around? Given the singledispatch we have here, I think we should show the different behaviour depending on the input.